js/noms/src/http-batch-store.js

```diff
@@ -94,14 +98,16 @@ export class Delegate {
   _rpc: RpcStrings;
   _readBatchOptions: FetchOptions;
   _writeBatchOptions: FetchOptions;
-  _rootOptions: FetchOptions;
+  _getRootOptions: FetchOptions;
+  _updateRootOptions: FetchOptions;
   _body: ArrayBuffer;
 
   constructor(rpc: RpcStrings, fetchOptions: FetchOptions) {
     this._rpc = rpc;
     this._readBatchOptions = mergeOptions(readBatchOptions, fetchOptions);
     this._writeBatchOptions = mergeOptions(writeBatchOptions, fetchOptions);
-    this._rootOptions = fetchOptions;
+    this._getRootOptions = fetchOptions;
+    this._updateRootOptions = mergeOptions(updateRootOptions, fetchOptions);
```

Comment:
Any reason not to share the options for write and updateRoot? Seems unlikely that they would be different.